### ISSUE TITLE
[Raspberry Pi OS (Bullseye)] arm | Kodi missing inputstream.ffmpegdirect

### ISSUE THREAD

**Issue by ReWiG** [xbmc/inputstream.ffmpegdirect#157]
Is there any inputstream.ffmpegdirect addon for Linux version?
I am using a freshly installed version of Kodi 19.3 from the official Raspberry Pi OS Bullseye repository. In the app store, I just don't have the inputstream.ffmpegdirect extension, although it is a dependency for the InputStream Adaptive and without it, I keep getting an error when starting the broadcast. 
`uname -a
Linux kodi 5.10.83-v7l+ #1499 SMP Tue Dec 7 14:08:09 GMT 2021 armv7l GNU/Linux
`

**Comment by phunkyfish** [comment]
That is up to whomever maintains your package repository. Linux repos are not maintained by team-kodi outside of a Ubuntu PPA.

**Comment by ReWiG** [comment]
> That is up to whomever maintains your package repository. Linux repos are not maintained by team-kodi outside of a Ubuntu PPA.

But this is a Kodi add-on repository, not an operating system repository. I would happily compile this add-on myself, but it doesn't work for me, I follow the instructions in the README, but I get the following error:

> -- [download 95% complete]
-- [download 96% complete]
-- [download 97% complete]
-- [download 98% complete]
-- [download 99% complete]
-- [download 100% complete]
-- verifying file...
       file='/home/pi/Исходники и deb/te/inputstream.ffmpegdirect/build/build/download/gmp-6.2.0.tar.xz'
-- Downloading... done
-- extracting...
     src='/home/pi/Исходники и deb/te/inputstream.ffmpegdirect/build/build/download/gmp-6.2.0.tar.xz'
     dst='/home/pi/Исходники и deb/te/inputstream.ffmpegdirect/build/build/gmp/src/gmp'
-- extracting... [tar xfz]
-- extracting... [analysis]
-- extracting... [rename]
-- extracting... [clean up]
-- extracting... done
[  3%] No update step for 'gmp'
[  4%] Performing patch step for 'gmp'
[  5%] Performing configure step for 'gmp'
/bin/sh: 1: PKG_CONFIG_PATH=/home/pi/Исходники и deb/te/inputstream.ffmpegdirect/build/build/depends/lib/pkgconfig: not found
make[2]: *** [CMakeFiles/gmp.dir/build.make:128: build/gmp/src/gmp-stamp/gmp-configure] Ошибка 127
make[1]: *** [CMakeFiles/Makefile2:335: CMakeFiles/gmp.dir/all] Ошибка 2
make: *** [Makefile:103: all] Ошибка 2

**Comment by phunkyfish** [comment]
Yes, the addon repo you speak of is not managed by team kodi. But as far as I’m aware this is all in the Debian OS repo.

@basilgello should be able to advise.

**Comment by ReWiG** [comment]
What can you advise about the build error?

**Comment by phunkyfish** [comment]
If you are on Linux you should not be building any libraries as far as I’m aware. Debian should skip that part of the build.

@basilgello will know as he maintains the kodi packages in Debian.

**Comment by basilgello** [comment]
@ReWiG You can install pkgconfig but it is honestly interesting to see `apt policy kodi kodi-inputstream-ffmpegdirect` first.

**Comment by ReWiG** [comment]
> apt policy kodi kodi-inputstream-ffmpegdirect

Of course there he is 

```
sudo apt policy kodi kodi-inputstream-ffmpegdirect
kodi:
  Установлен: 2:19.3-1~bullseye
  Кандидат:   2:19.3-1~bullseye
  Таблица версий:
 *** 2:19.3-1~bullseye 500
        500 http://archive.raspberrypi.org/debian bullseye/main armhf Packages
        100 /var/lib/dpkg/status
kodi-inputstream-ffmpegdirect:
  Установлен: (отсутствует)
  Кандидат:   (отсутствует)
  Таблица версий:
```

**Comment by ReWiG** [comment]
> You can install pkgconfig

If you are talking about the "pkg-config" package, then I already have it installed (0.29.2-1)

**Comment by basilgello** [comment]
@ReWiG You can go past this error if your build path has no spaces: `Исходники и deb`.

However, I'd raise the big on Raspberry OS forums, because Raspberry OS is not a vanilla Debian.

**Comment by ReWiG** [comment]
> You can go past this error if your build path has no spaces: `Исходники и deb`.

Thanks, it helped me and I was able to compile the extension for myself. It's a shame that it is not possible to add it to the add-ons repository, I created a thread on the Raspberry forum, but I strongly doubt that anyone will do it there. Apparently, Raspberry Pi users will have to endure errors when watching IPTV in the IPTV simple client add-on. Well, if nothing else can be done, then you can close this topic.

**Comment by robiwano** [comment]
I'm just building it myself. @ReWiG you never managed to install it ? Have you tried `make install` after building? (I haven't tried it yet)

**Comment by basilgello** [comment]
Proper installation is to install the resulting ZIP file, not `make install`-ing every add-on.

**Comment by basilgello** [comment]
> Well, if nothing else can be done, then you can close this topic.

I think the only way is to reach Raspberry OS maintainer, whose,email I am trying to find now. Actually, nothing is required from them apart from a no-chande build.

**Comment by basilgello** [comment]
@spl237 maybe you can help with this issue?

**Comment by robiwano** [comment]
> Proper installation is to install the resulting ZIP file, not `make install`-ing every add-on.

Through the Kodi interface, right?

**Comment by basilgello** [comment]
> Through the Kodi interface, right?

Correct! https://kodi.wiki/view/Archive:Install_add-ons_from_zip_files

**Comment by robiwano** [comment]
Ok, in the .../xbmc/build/addons folder I had to zip up the inputstream.ffmpegdirect folder like so:
`> cd /home/pi/src/xbmc/build/addons`
`> zip -r inputstream.ffmpegdirect.zip inputstream.ffmpegdirect/*`

Then the installation worked fine!

**Comment by basilgello** [comment]
> zip -r inputstream.ffmpegdirect.zip inputstream.ffmpegdirect/*

I use `zip -r inputstream.ffmpegdirect.zip --symlinks inputstream.ffmpegdirect/*`

**Comment by lrusak** [comment]
On Linux you don't need to use the zip. you can just copy the directory to the installation location or use make install

**Comment by robiwano** [comment]
Tried make install but there is no install target.

**Comment by boucchar** [comment]
Hi, i successfuly build inputstream.ffmpegdirect to my rpi4 bullseye os. I copy the build folder to /usr/share/kodi/addons and kodi 19.3 detect the addon. It demand to activate, i choose yes. But unfortunatly, after click yes, kodi say that this version is not compatible. So, is it supposed to work? Inputstreamm.ffmpegdirect seem to be at version 19.1 and kodi at version 19.3... is it problematic? 

Thank you for your support!

**Comment by phunkyfish** [comment]
There is no 19.1 of ffmpegdirect for Matrix.

What is the latest commit of the repo you cloned?

**Comment by boucchar** [comment]
Sorry, not 19.1 i mean 19.0.1. I follow the build instruction provided by you on Github. So, i clone with : git clone --branch Matrix https://github.com/xbmc/inputstream.ffmpegdirect.git

**Comment by phunkyfish** [comment]
Can you provide a listing of the folder you copied to addons?

Also a debug log would be useful.

**Comment by boucchar** [comment]
Finally, i restart from a clean install of rpios bullseye (rpi4), sudo apt update + upgrade, clean kodi 19.3, remake the build process of inputstream.ffmpegdirect and it work! Maybe i made something wrong in my first attempt, idon't know... but currently it work like a charm. For infos, i need inputstream.ffmpegdirect as a dependency for kodi-pvr-nextpvr. Thank you for your work and your support.

**Comment by phunkyfish** [comment]
> Finally, i restart from a clean install of rpios bullseye (rpi4), sudo apt update + upgrade, clean kodi 19.3, remake the build process of inputstream.ffmpegdirect and it work! Maybe i made something wrong in my first attempt, idon't know... but currently it work like a charm. For infos, i need inputstream.ffmpegdirect as a dependency for kodi-pvr-nextpvr. Thank you for your work and your support.

No problem, glad you got it working. Hopefully the package repo maintainer can include it soon so others don't need to do this to get things working.

**Comment by N3oTraX** [comment]
> Finally, i restart from a clean install of rpios bullseye (rpi4), sudo apt update + upgrade, clean kodi 19.3, remake the build process of inputstream.ffmpegdirect and it work! Maybe i made something wrong in my first attempt, idon't know... but currently it work like a charm. For infos, i need inputstream.ffmpegdirect as a dependency for kodi-pvr-nextpvr. Thank you for your work and your support.

Hi, is-it possible to host a zipped version of your build to share it with me. I tried to compile it on mine but it failed. I don't want to make a fresh install because all my stuff is already on it ^^"

**Comment by boucchar** [comment]
> > Finally, i restart from a clean install of rpios bullseye (rpi4), sudo apt update + upgrade, clean kodi 19.3, remake the build process of inputstream.ffmpegdirect and it work! Maybe i made something wrong in my first attempt, idon't know... but currently it work like a charm. For infos, i need inputstream.ffmpegdirect as a dependency for kodi-pvr-nextpvr. Thank you for your work and your support.
> 
> Hi, is-it possible to host a zipped version of your build to share it with me. I tried to compile it on mine but it failed. I don't want to make a fresh install because all my stuff is already on it ^^"

Hi, yes it is possible: https://os5.mycloud.com/action/share/cf55f05d-c7ab-4f28-af58-3a012ec3d95d

**Comment by N3oTraX** [comment]
> > > Finally, i restart from a clean install of rpios bullseye (rpi4), sudo apt update + upgrade, clean kodi 19.3, remake the build process of inputstream.ffmpegdirect and it work! Maybe i made something wrong in my first attempt, idon't know... but currently it work like a charm. For infos, i need inputstream.ffmpegdirect as a dependency for kodi-pvr-nextpvr. Thank you for your work and your support.
> > 
> > 
> > Hi, is-it possible to host a zipped version of your build to share it with me. I tried to compile it on mine but it failed. I don't want to make a fresh install because all my stuff is already on it ^^"
> 
> Hi, yes it is possible: https://os5.mycloud.com/action/share/cf55f05d-c7ab-4f28-af58-3a012ec3d95d

Many thanks, it worked like a charm, you saved me :D

**Comment by no1be4me** [comment]
Thanks for sharing your work 
https://os5.mycloud.com/action/share/cf55f05d-c7ab-4f28-af58-3a012ec3d95d 
I too tried compiling it and it failed.

**Comment by zwejst3in** [comment]
Thanks for sharing your work
https://os5.mycloud.com/action/share/cf55f05d-c7ab-4f28-af58-3a012ec3d95d

made my day easy today ;)

**Comment by vadd98** [comment]
Hi, may you share the package again? The link https://os5.mycloud.com/action/share/cf55f05d-c7ab-4f28-af58-3a012ec3d95d is broken

**Comment by boucchar** [comment]
Hi, new link: https://os5.mycloud.com/action/share/a2ba5e2d-76db-477c-8616-a58407d08223
